pkg/action/list.go

```diff
@@ -159,14 +159,21 @@ func (l *List) Run() ([]*release.Release, error) {
 		}
 	}
 
-	results, err := l.cfg.Releases.List(func(rel *release.Release) bool {
+	filterFunction := func(rel *release.Release) bool {
```

Comment:
Now (since we are targeting Helm v4, and do not need to maintain API backwards compatibility), we can just pass the label sector to `List(...)`, and avoid the extra complexity of  `CanPushDownLabelSelector(...)` and `ListWithSelector(...)`.

ie., just do the below and fold the selector logic into `List(..)` itself (where applicable, based on the driver):
```
	results, err := l.cfg.Releases.List(l.Selector, func(rel *release.Release) bool {
```